src/lib/reasoners/satml_frontend.ml

```diff
@@ -1398,14 +1398,17 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
     match E.type_info t with
     | Ty.Tbool ->
       begin
-        match ME.find_opt t env.gamma with
-        | None ->
-          begin
-            match ME.find_opt (E.neg t) env.gamma with
-            | None -> None
-            | Some _ -> Some E.faux
-          end
-        | Some _ -> Some E.vrai
+        let bmodel = SAT.boolean_model env.satml in
+        List.find_map
+          (fun Atom.{lit; neg = {lit=neglit; _}; _} ->
```

Comment:
Nit: it would be better to call `Atom.literal` and `Expr.neg` here instead; the structure of atoms should be treated as an abstract type as much as possible.

Reply:
Is there something preventing us from defining Atom.t as an abstract type?

Reply:
I believe both `Satml_types` and `Satml` modules actually sort of need to know its internals currently, so nothing major as far as I know. Making it an abstract type is not entirely trivial refactoring, but we can try to preemptively not make it too hard on ourselves for when we actually do it :)

Reply:
Please open an low-priority issue on this point ;)